dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -305,151 +288,78 @@ def predict_fn_scores(self, input_instance):
+        return proximity_loss
 
     def compute_loss(self, cfs, desired_range, desired_class):
         """Computes the overall loss"""
         self.yloss = self.compute_yloss(cfs, desired_range, desired_class)
         self.proximity_loss = self.compute_proximity_loss(cfs) if self.proximity_weight > 0 else 0.0
-        self.diversity_loss = self.compute_diversity_loss(cfs) if self.diversity_weight > 0 else 0.0
-        # TODO this is not needed for label encoding
-        #self.regularization_loss = self.compute_regularization_loss(cfs)
-
-
-        self.loss = self.yloss + (self.proximity_weight * self.proximity_loss) + (
-                    self.diversity_weight * self.diversity_loss)
-
+        self.loss = np.reshape(np.array(self.yloss + (self.proximity_weight * self.proximity_loss)), (-1, 1))
+        index = np.reshape(np.arange(len(cfs)), (-1, 1))
+        self.loss = np.concatenate([index, self.loss], axis=1)
         return self.loss
 
     def mate(self, k1, k2, features_to_vary, query_instance):
         """Performs mating and produces new offsprings"""
         # chromosome for offspring
-        child_chromosome = []
-        for i in range(self.total_CFs):
-            # temp_child_chromosome = []
-            one_init = [[]]
-            for j in range(len(self.data_interface.feature_names)):
-            #for jx, (gp1, gp2) in enumerate(zip(k1[i][0], k2[i][0])):
-                gp1 = k1[i][0][j]
-                gp2 = k2[i][0][j]
-                feat_name = self.data_interface.feature_names[j]
-
-                # random probability
-                prob = random.random()
-
-                # if prob is less than 0.45, insert gene from parent 1
-                if prob < 0.45:
-                    one_init[0].append(gp1)
-
-                # if prob is between 0.45 and 0.90, insert gene from parent 2
-                elif prob < 0.90:
-                    one_init[0].append(gp2)
-
-                #otherwise insert random gene(mutate) for maintaining diversity
-                else:
-                    if feat_name in features_to_vary:
-                        if feat_name in self.data_interface.continuous_feature_names:
-                            one_init[0].append(np.random.uniform(self.feature_range[feat_name][0], self.feature_range[feat_name][0]))
-                        else:
-                            one_init[0].append(np.random.choice(self.feature_range[feat_name]))
-                    else:
-                        one_init[0].append(query_instance[0][j])
+        one_init = np.zeros(self.data_interface.number_of_features)
+        for j in range(self.data_interface.number_of_features):
+            gp1 = k1[j]
+            gp2 = k2[j]
+            feat_name = self.data_interface.feature_names[j]
 
-            child_chromosome.append(np.array(one_init))
-        return child_chromosome
+            # random probability
+            prob = random.random()
```

Comment:
for consistency, can use np.random and avoid importing python's default random. They have different seeds.